### PR TITLE
Implement str.isnumeric

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the PR. It's currently failing `flake8` https://dev.azure.com/numba/numba/_build/results?buildId=3164&view=logs&jobId=687448c2-4012-5c9e-43b3-3520c9635a30&taskId=dba03d4a-57a7-5e59-a47a-79fe530f297a&lineStart=64&lineEnd=65&colStart=1&colEnd=1

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -1332,6 +1332,25 @@ def test_lower(self):
...
+        cpython = ['', 'a', '0', '\u2460', '\xbc', '\u0660', '0123456789',
+                   '0123456789a', '\U00010401', '\U00010427', '\U00010429',
+                   '\U0001044E', '\U0001F40D', '\U0001F46F', '\U00011065',
+                   '\U0001D7F6', '\U00011066', '\U000104A0', '\U0001F107']
+        # https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Lib/test/test_unicode.py#L742-L749    # noqa: E501
+        cpython_extras = ['\uD800', '\uDFFF', '\uD800\uD800', '\uDFFF\uDFFF',
+                          'a\uD800b\uDFFF', 'a\uDFFFb\uD800', 'a\uD800b\uDFFFa',
+                          'a\uDFFFb\uD800a']
+
+        msg = 'Results of "{}".isspace() must be equal'
```
Comment (stuartarchibald) [inline numba/tests/test_unicode.py]:
```suggestion
        msg = 'Results of "{}".isnumeric() must be equal'
```

File: numba/unicode.py [hunk 2]
```diff
@@ -1361,6 +1361,25 @@ def impl(a):
...
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12096-L12124    # noqa: E501
+@overload_method(types.UnicodeType, 'isnumeric')
+def unicode_isnumeric(data):
+    """Implements UnicodeType.isnumeric()"""
+
+    def impl(data):
+        length = len(data)
+        if length == 0:
```
Comment (stuartarchibald) [inline numba/unicode.py]: Single char fast path ? 
https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12110-L12113

File: numba/unicode_support.py [hunk 3]
```diff
@@ -217,7 +217,8 @@ def _PyUnicode_IsDigit(ch):
 
 @register_jitable
 def _PyUnicode_IsNumeric(ch):
-    raise NotImplementedError
+    ctype = _PyUnicode_gettyperecord(ch)
```
Comment (stuartarchibald) [inline numba/unicode_support.py]: Github reference with SHA?

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch, thanks.

**Comment by Rubtsowa** [comment]
Conflicts resolved. Do you have anymore comment or recommendations related to this PR?

**Comment by Rubtsowa** [comment]
Сhange the label, please.

**Comment by esc** [comment]
@Rubtsowa thanks for the ping on this PR, I have updated the label accordingly.

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts by merging master into this branch. Thanks!

**Comment by Rubtsowa** [comment]
@stuartarchibald, I merged conflicts.

**Comment by Rubtsowa** [comment]
@stuartarchibald, I resolved conflicts.

**Comment by Rubtsowa** [comment]
Please check this PR! If everything is correct, please merge.

**Pull request closed without merging**